### PR TITLE
[feature/ext-browser-tokenauth] Token authorization by external browser

### PR DESCRIPTION
**Pull request by felix-schwarz**
## Description
Allows using an external browser with custom URL schemes to handle token authorization.

## Related Issue
https://github.com/owncloud/enterprise/issues/4256

## How Has This Been Tested?
On device with the [MobileIron Web@Work](https://apps.apple.com/app/id596170970) browser.

## Screenshots (if appropriate):
what users see when prematurely switching back from the browser | browser session
--|--
Bildschirmfoto 2021-05-05 um 15 00 11 | Bildschirmfoto 2021-05-05 um 15 09 41

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING**](https://github.com/owncloud/ios-app/blob/master/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] Added changelog files for the fixed issues in folder changelog/unreleased

### CODE REVIEW

File: ownCloud/Resources/Theming/Branding.plist [hunk 1]
```diff
@@ -1,5 +1,8 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
 <plist version="1.0">
-<dict/>
+<dict>
+	<key>authentication.browser-session-class</key>
+	<string>MIBrowser</string>
```
Comment (hosy) [inline ownCloud/Resources/Theming/Branding.plist:6]: Example should be removed. I will put the sample code inside the PR comment.

**Review by hosy — CHANGES_REQUESTED** [review]

**Comment by hosy** [comment]
@jesmrec for testing please put this into `Branding.plist`:

```
<?xml version="1.0" encoding="UTF-8"?>
 <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
 <plist version="1.0">
 <dict>
 	<key>authentication.browser-session-class</key>
 	<string>MIBrowser</string>
 </dict>
 </plist>
```

File: ownCloud/Resources/Theming/Branding.plist (hunk 1 above)
Comment (felix-schwarz) [reply ownCloud/Resources/Theming/Branding.plist]: Done. Turned the the plist into an example at `ownCloud/Resources/Theming/Branding-mibrowser-example.plist`.

**Comment by felix-schwarz** [comment]
@jesmrec For convenience / reference, the plist shared by @hosy now also is an example, to be found at `ownCloud/Resources/Theming/Branding-mibrowser-example.plist`.

**Review by hosy — APPROVED** [review]

**Comment by jesmrec** [comment]
Checks performed:

- [X] OAuth2 with MobileIron Web@Work -> correct auth
- [X] OIDC with MobileIron Web@Work -> correct auth
- [X] OAuth2 without MobileIron Web@Work -> correct error
- [X] OIDC without MobileIron Web@Work -> correct error

It's approved by using 

```
<?xml version="1.0" encoding="UTF-8"?>
 <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
 <plist version="1.0">
 <dict>
 	<key>authentication.browser-session-class</key>
 	<string>MIBrowser</string>
 </dict>
 </plist>
```

mentioned above.

I assume that this is an specific case (MIBrowser) that only will be available in customer scenarios (`Branding.plist` needs specific set up). 

Only the tests listed above were done.

**Comment by hosy** [comment]
@jesmrec please can you check if `owncloud://` for opening private links and local file ids is still working. This function may be affected.

**Comment by jesmrec** [comment]
Performed some checks regarding universal links:

- [X] Universal link to file in root 
- [X] Universal link to file in non-root
- [X] Universal link to folder in root
- [X] Universal link to folder in non-root
- [X] Universal link to item in different account
- [X] Universal link to non-existing id

Tested with iPhoneXR v14.4
commit `bbfa68ddd`

**Review by jesmrec — APPROVED** [review]

**Pull request merged**